src/basic/cgroup-util.c

```diff
@@ -2249,6 +2249,26 @@ bool cg_is_legacy_wanted(void) {
         return !cg_is_unified_wanted();
 }
 
+int cg_weight_parse(const char *s, uint64_t *ret) {
+        uint64_t u;
+        int r;
+
+        if (isempty(s)) {
+                *ret = CGROUP_WEIGHT_INVALID;
+                return 0;
+        }
+
+        r = safe_atou64(s, &u);
+        if (r < 0)
+                return r;
+
+        if (u < CGROUP_WEIGHT_MIN || u > CGROUP_WEIGHT_MAX)
```

Comment:
It's testing whether the value is out of range, so < MIN || > MAX is the right test. CGROUP_WEIGHT_IS_OK() would allow CGROUP_WEIGHT_INVALID value as a valid input, which isn't quite what we want here.